db/commitlog/commitlog.cc

```diff
@@ -1512,28 +1512,42 @@ void db::commitlog::segment_manager::flush_segments(uint64_t size_to_remove) {
         high = replay_position(high.id + 1, 0);
     }
 
-    auto n = size_to_remove;
+    // Now get a set of used CF ids:
+    std::unordered_set<cf_id_type> ids;
 
-    if (size_to_remove != 0) {
-        for (auto& s : _segments) {
-            if (n <= s->size_on_disk()) {
-                high = replay_position(s->_desc.id, db::position_type(s->size_on_disk()));
-                break;
-            }
-            n -= s->size_on_disk();
+    uint64_t n = size_to_remove;
+    uint64_t flushing = 0;
+
+    for (auto& s : _segments) {
+        if (s->is_still_allocating()) {
```

Comment:
This seems to be a behavioral change and it is unclear why we're breaking from the loop here.

Reply:
It is not really, just applying the same logic as (previously) above w.r. high rp. Still allocating segments cannot be released, i.e. is pointless to include in a flush request. The allocating segment will also always be the last one, so in effect we do almost always end at end()-1, unless things are shut/shutting down, in which case we could have _no_ allocating segment.

Reply:
@elcallio so let's add some comments explaining those decisions in the code.
As someone that isn't intimately familiar with the implementation I couldn't find my way around...

Reply:
Otherwise LGTM

Reply:
Fixed.